### PR TITLE
fix(dist): avoid var PRELOAD cover environmnet vars

### PR DESCRIPTION
**Pull request by aroundabout**
## Purpose of the PR

- fix bug in #2241 
  - In start-hugegraph.sh, pre-declared var PRELOAD will cover environment PRELOAD
  - extract function "exit_with_usage_help" in util.sh
  - original issue: feature: pre-load some data in container(subtask of https://github.com/apache/incubator-hugegraph/issues/840)

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [x] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:
    - xxx

## Does this PR potentially affect the following parts?

- [x]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects (typed here)

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [x]  `Doc - No Need`

### CODE REVIEW

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh [hunk 1]
```diff
@@ -18,12 +18,30 @@
...
 
+function abs_path() {
+    SOURCE="${BASH_SOURCE[0]}"
+    while [[ -h "$SOURCE" ]]; do
+        DIR="$(cd -P "$(dirname "$SOURCE")" && pwd)"
+        SOURCE="$(readlink "$SOURCE")"
+        [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE"
+    done
+    cd -P "$(dirname "$SOURCE")" && pwd
+}
```
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:34]: shall we move them to `util.sh`? (adjust the position)

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh [hunk 2]
```diff
@@ -35,45 +53,18 @@ while getopts "d:g:m:p:s:j:t:v" arg; do
         t) SERVER_STARTUP_TIMEOUT_S="$OPTARG" ;;
         # TODO: should remove it in future (check the usage carefully)
         v) VERBOSE="verbose" ;;
-        ?) echo "USAGE: $0 [-d true|false] [-g g1] [-m true|false] [-p true|false] [-s true|false] [-j java_options]
-                [-t timeout]" && exit 1 ;;
+        ?) exit_with_usage_help ;;
```
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:56]: add a `WARN`comment  to update the usage info (if modify the options)

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh [hunk 3]
```diff
@@ -18,12 +18,30 @@
 OPEN_MONITOR="false"
 OPEN_SECURITY_CHECK="true"
 DAEMON="true"
-PRELOAD="false"
```
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:21]: test it in binary package when we start with `bin/start-hugegraph.sh -p true` (if we remove it)

Comment (aroundabout) [reply hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:21]: Tested.

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh [hunk 4]
```diff
@@ -18,12 +18,30 @@
...
+
+BIN=$(abs_path)
+TOP="$(cd "$BIN"/../ && pwd)"
+CONF="$TOP/conf"
+LOGS="$TOP/logs"
+SCRIPTS="$TOP/scripts"
+PID_FILE="$BIN/pid"
+
+. "$BIN"/util.sh
+
```
[the remark this answers is not in the document]
Comment (imbajin) [reply hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:44]: if no reply > 4d, u could replace it & test it by yourself first

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh [hunk 5]
```diff
@@ -18,12 +18,30 @@
 OPEN_MONITOR="false"
 OPEN_SECURITY_CHECK="true"
 DAEMON="true"
-PRELOAD="false"
 #VERBOSE=""
 GC_OPTION=""
 USER_OPTION=""
 SERVER_STARTUP_TIMEOUT_S=30
```
Comment (aroundabout) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:25]:
```suggestion
# todo: move abs_path funtion to shell like util.sh
```

**Review by imbajin — APPROVED** [review]

**Pull request merged**